src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/DefaultVSCredentialServiceProvider.cs

```diff
@@ -87,7 +87,10 @@ public async Task<NuGet.Configuration.ICredentialService> GetCredentialServiceAs
             }
 
             // Initialize the credential service.
-            var credentialService = new CredentialService(new AsyncLazy<IEnumerable<ICredentialProvider>>(() => System.Threading.Tasks.Task.FromResult((IEnumerable<ICredentialProvider>)credentialProviders)), nonInteractive: false, handlesDefaultCredentials: PreviewFeatureSettings.DefaultCredentialsAfterCredentialProviders);
+            var credentialService = new CredentialService(
+                new AsyncLazy<IEnumerable<ICredentialProvider>>(() => System.Threading.Tasks.Task.FromResult((IEnumerable<ICredentialProvider>)credentialProviders)),
+                nonInteractive: KnownUIContexts.CloudEnvironmentConnectedContext.IsActive, // cannot only interact when cloudconnection is not active.
```

Comment:
I don't know if this is the right thing to check. 
I think this is only true when there's a client connected to the server. (maybe only client side)
You might be looking for something like https://github.com/NuGet/NuGet.Client/blob/e3c62df98bec2b6232c0cec93cb60b5fa1d9aa95/src/NuGet.Clients/NuGet.Console/OutputConsole/OutputConsoleProvider.cs#L123-L130 instead. 

I could be wrong though.

Reply:
likely we'll need that several places in our code. 
options: 
1) call VS method:  Microsoft.VisualStudio.Cascade.CommonUtility.IsVSOnlineServerAsync()
2) move method from OutputConsoleProvider to NuGet.Common somewhere (or where do we keep things like this?)
3) copy the code

Thoughts?

I lean t

Reply:
NuGet.VisualStudio.Common likely. 

1 or 2 are both fine I think. You can use 1 if we have access to it. For 2, I think you likely want NuGet.VisualStudio.Common.

Reply:
i'm prototyping num 2. moving our helper to NuGet.VisualStudio.Common.dll - NuGet.VisualStudio.VisualStudioContextHelper class - IsVSOnlineServerAsync method moved from OutputConsoleProvider.
Thoughts?

Reply:
I think that's fine.